spring-cloud-gateway-server-mvc/src/main/java/org/springframework/cloud/gateway/server/mvc/filter/BeforeFilterFunctions.java

```diff
@@ -407,7 +412,7 @@ public static Function<ServerRequest, ServerRequest> stripPrefix(int parts) {
 
 			URI prefixedUri = UriComponentsBuilder.fromUri(request.uri())
 				.replacePath(newPath.toString())
-				.build()
+				.build(true)
```

Comment:
The stripped path is already encoded, so no additional encoding will be applied. 
Furthermore, other parts such as request parameters are not affected.